src/app.ts

```diff
@@ -3,14 +3,20 @@ import express from 'express';
 import pinoHttp from 'pino-http';
 import cors from 'cors';
 import { rootRouter } from './routers';
-import { errorHandler, processJwt, addUserContext } from './middleware';
+import {
+	errorHandler,
+	processJwt,
+	addUserContext,
+	addRoleContext,
+} from './middleware';
 import { getLogger } from './logger';
 
 const logger = getLogger(__filename);
 const app = express();
 app.use(cors());
 app.use(processJwt);
 app.use(addUserContext);
```

Comment:
Suppose I reversed this line `app.use(addUserContext);` with `app.use(addRoleContext);` and removed `app.use(processJwt);` what happens? A nice error: 401, "No authorization token was found." Good! This is probably old functionality but something I hadn't tried yet. Better late than never.

I see that each step builds on the previous so the order is important. `processJwt` adds `auth` to the request, `addUserContext` adds `user` to the request (requiring `auth` presence), and `addRoleContext` adds `role` to the request (requiring `user` presence). Right?